lib/ipmi_user.c

```diff
@@ -477,16 +481,31 @@ ipmi_user_build_password_prompt(uint8_t user_id)
  *
  * @returns pointer to char with password
  */
-char *
-ask_password(uint8_t user_id)
+int
+ask_password(uint8_t user_id, char *buf, size_t buflen)
 {
-	const char *password_prompt =
-		ipmi_user_build_password_prompt(user_id);
-# ifdef HAVE_GETPASSPHRASE
```

Comment:
Correct me if I am wrong.

`getpassphrase(3)` is a Solaris only API?
`termios(3)` is recommended instead of `getpass(3)` on Linux according to the man page, however I believe that mainly to be true due to the ease of introducing bugs such as the one I was attempting to fix here. The getpass implementation just uses termios under the hood and uses a static buffer (foolishly?) rather than taking a buffer as an arg, so the code is mostly the same anyways.
`getpass` is not depricated in the BSD manpages (FreeBSD specifically), however they have their own passphrase API `readpassphrase(3)`

I decided to continue the use of getpass and remove the Solaris API to first simply the code, and also maintain some portability without if def cpp all over the place.

Will await further instruction.

Reply:
Well, although I'm not sure Solaris compatibility is still maintained otherwise, I see that Solaris is still around (11.4 was released 2 years ago and in still maintained), and some people may be still using ipmitool for it.

Anyway, if Solaris support is to be dropped, that must be done explicitly with a separate PR affecting all the sources, not just one file. So please restore the `getpassphrase()` support here unless you're sure that getpass() can work for Solaris and, for instance, will not truncate the password to 9 characters like it says [here](https://docs.oracle.com/cd/E23823_01/html/816-5168/getpass-3c.html).

Reply:
Yep that is fair. I will do my best there, I don't have a means of testing Solaris specifically, but ill reference the man page. I'll update this PR asap. Thanks.

Reply:
Reading the man page, getpass on Solaris only supports 9 bytes, so that makes sense why getpassphrase is needed. Fixing in my latest rev.

Reply:
Thanks. Will wait for the termios version then.